From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Otto=20Pfl=C3=BCger?= <otto.pflueger@abscue.de>
Date: Mon, 16 Jun 2025 22:16:38 +0200
Subject: [PATCH 042/225] power: supply: sc27xx: fix handling of negative
 CURRENT_NOW values
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use a signed integer variable for the ADC value so that it is correctly
converted to s64 when passed as an argument to sc27xx_fgu_adc_to_current.

Signed-off-by: Otto Pflüger <otto.pflueger@abscue.de>
---
 drivers/power/supply/sc27xx_fuel_gauge.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/power/supply/sc27xx_fuel_gauge.c b/drivers/power/supply/sc27xx_fuel_gauge.c
index 59249cf92..580bd2cd2 100644
--- a/drivers/power/supply/sc27xx_fuel_gauge.c
+++ b/drivers/power/supply/sc27xx_fuel_gauge.c
@@ -403,8 +403,7 @@ static int sc27xx_fgu_get_vol_now(struct sc27xx_fgu_data *data, int *val)
 
 static int sc27xx_fgu_get_cur_now(struct sc27xx_fgu_data *data, int *val)
 {
-	int ret;
-	u32 cur;
+	int ret, cur;
 
 	ret = regmap_read(data->regmap, data->base + SC27XX_FGU_CURRENT_BUF,
 			  &cur);
